iio/iio_demo/demo_dev.c

```diff
@@ -53,6 +53,56 @@
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 
+/* Read a device register. The register address to read is set on
+ * in desc->active_reg_addr in the function set_demo_reg_attr
+ */
+ssize_t get_demo_reg_attr(void *device, char *buf, size_t len,
+			  const struct iio_ch_info *channel)
+{
+	struct iio_demo_desc	*desc = device;
+	uint32_t		value;
+
+	value = desc->dummy_regs[desc->active_reg_addr];
+
+	return snprintf(buf, len, "%"PRIu32"", value);
+}
+
+/* Flow of reading and writing registers. This is how iio works for
+ * direct_reg_access attribute:
+ * Read register:
+ * 	   //Reg_addr in decimal
+ * 	   reg_addr = "10";
+ * 	1. set_demo_reg_attr(dev, reg_addr, len, channel);
+ * 	2. get_demo_reg_attr(dev, out_buf, out_len, channel);
+ * Write register:
+ * 	   sprintf(write_buf, "0x%x 0x%x", reg_addr, value);
+ * 	1. set_demo_reg_attr(dev, write_buf, len, channel);
+ */
+ssize_t set_demo_reg_attr(void *device, char *buf, size_t len,
+			  const struct iio_ch_info *channel)
+{
+	struct iio_demo_desc	*desc = device;
+	uint32_t		nb_filled;
+	uint32_t		addr;
+	uint32_t		value;
+
+	nb_filled = sscanf(buf, "0x%x 0x%x", &addr, &value);
+	if (nb_filled == 2) {
+		/* Write register */
+		desc->dummy_regs[addr] = value;
```

Comment:
`addr` should be somehow limited to a maximum value, otherwise this can be used to read memory that shouldn't be available to the IIO client user.

Reply:
done